tests/utils/service.spec.tsx

```diff
@@ -0,0 +1,55 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+import { Project, Directory, File, FileType } from "../../src/model";
+import { Service, Language } from "../../src/service";
+import * as compilerServices from "../../src/compilerServices";
+
+class TestService {
+  public input;
+  async compile(input) {
+    this.input = input;
+    const items = { "a.txt": { content: "text" } };
+    return { success: true, items, };
+  }
+}
+
+function generateFiles() {
+    const project = new Project();
+    const dir = new Directory("dir");
+    project.addFile(dir);
+    const subdir = new Directory("subdir");
+    dir.addFile(subdir);
+    const a = new File("a.js", FileType.JavaScript);
+    project.addFile(a);
+    const b = new File("main.ts", FileType.HTML);
+    dir.addFile(b);
+    const c = new File("c.md", FileType.Markdown);
+    subdir.addFile(c);
+    return { a, b, c, };
+}
+
+describe("Service.compileFiles tests", () => {
+  let saveCCS;
+  beforeAll(() => {
+    saveCCS = (compilerServices as any).createCompilerService;
+  });
+  afterAll(() => {
+    (compilerServices as any).createCompilerService = saveCCS;
```

Comment:
@ollelauribostrom I am not sure if there is something better than doing restoring mocks this way.

Reply:
You could probably use jest.mock instead, so something like,

```js
const compile = jest.fn();

jest.mock("../../src/compilerServices", () => ({
  createCompilerService: () => ({
    compile: compile.mockImplementation(() => ({
      success: true,
      items: { "a.txt": { content: "text" } }
    }))
  }),
  Language: {
    JavaScript: "javascript"
  }
}));

import { Project, Directory, File, FileType } from "../../src/model";
import { Service, Language } from "../../src/service";
```

```js
expect(result).toEqual(...)
expect(compile).toHaveBeenCalledWith(...)
```

```js
afterAll(() => {
  jest.unmock("../../src/compilerServices");
});
```